src/shared/components/buttons/primary.tsx

```diff
@@ -0,0 +1,40 @@
+import { StyleSheet, TouchableOpacity } from 'react-native';
+
+import { Typographies } from '../typographies';
+
+import { useStyles } from '~/shared/hooks/use-styles';
+import { createStylesheet } from '~/shared/styles/create-stylesheet';
+
+interface TextProps extends React.ComponentProps<typeof TouchableOpacity> {
+  text: string;
+  onPress?: () => void;
+  fillWidth: boolean;
```

Comment:
- `onPress` should be mandatory because a button always have an action.
- `fillWidth` should be false by default, then make it optional

_**same for other components**_